core/tauri/src/webview/mod.rs

```diff
@@ -568,17 +568,27 @@ tauri::Builder::default()
       }));
     }
 
-    if let Some(on_page_load_handler) = self.on_page_load_handler.take() {
-      let label = pending.label.clone();
-      let manager = manager.manager_owned();
-      pending
-        .on_page_load_handler
-        .replace(Box::new(move |url, event| {
-          if let Some(w) = manager.get_webview(&label) {
-            on_page_load_handler(w, PageLoadPayload { url: &url, event });
+    let label_ = pending.label.clone();
+    let manager_ = manager.manager_owned();
+    pending
+      .on_page_load_handler
+      .replace(Box::new(move |url, event| {
+        if let Some(w) = manager_.get_webview(&label_) {
+          if let PageLoadEvent::Finished = event {
+            match w.unlisten_all_js() {
+              Ok(_) => {
+                // println!("Clear all residual js listeners when page load.")
+              }
+              Err(e) => {
+                println!("Failed to clear residual js listeners when page load: {e}")
+              }
+            };
```

Comment:
Should be okay to ignore the error here.
```suggestion
            let _ = w.unlisten_all_js();
```